src/components/Registration/RegistrationForm.tsx

```diff
@@ -34,141 +30,193 @@ export const RegistrationForm = () => {
 
   const initialValues: IRegistration = {
     username: "",
-    firstname: "",
-    lastname: "",
     email: "",
     password: "",
     confirmpassword: "",
-  };
-
-  const onSubmit = (formValue: IRegistration) => {
-    const { username, firstname, lastname, email, password, confirmpassword } = formValue;
-    register(username, email, password, confirmpassword, firstname, lastname)
-    .then(
-      (res) => {
-        console.log("res z serwera", res)
-      },
-      (error) => {
-        console.log("res z serwera error", error)
-        console.log("nie działa");
-      },
-    );
+    firstname: "",
+    lastname: "",
   };
 
   return (
-    <Formik initialValues={initialValues} validationSchema={SignupSchema()} onSubmit={onSubmit}>
-      {(formValue) => {
+    <>
+    <Formik 
+      initialValues={initialValues} 
+      validationSchema={SignupSchema()} 
+      onSubmit={(formValue: IRegistration) => {
+        const { username, email, password, confirmpassword, firstname, lastname } = formValue;
+        register(username, email, password, confirmpassword, firstname, lastname).then(
+          () => {
+            setTimeout(() => {
+              navigate(paths.login, { replace: true })
+            }, 3000);
+            toast.success(t`toast.registration.success`)
+          },
+          (error) => {
+            console.log("error", error.response.status)
+            switch (error.response.status) {
+              case 400:
+                return toast.error(t`toast.registration.validation`);
+              default:
+                return toast.error(t`toast.registration.error`);
+            }
+          }
```

Comment:
można to zapisać w jednej linijce 
```js
({ response: { status } }) => toast.error(status === 400 ? t`toast.registration.validation` :  t`toast.registration.error`) 
```

Reply:
oooo, super rozwiązanie! nie pomyślałam w sumie. dzięki. już poprawione :)